tests/TestCase/Routing/Middleware/RoutingMiddlewareTest.php

```diff
@@ -133,4 +133,33 @@ public function testMissingRouteNotCaught()
         $middleware = new RoutingMiddleware();
         $middleware($request, $response, $next);
     }
+
+    /**
+     * Test route with _method being parsed correctly.
+     *
+     * @return void
+     */
+    public function testFakedRequestMethodParsed()
+    {
+        Router::connect('/articles-patch', [
+            'controller' => 'Articles',
+            'action' => 'index',
+            '_method' => 'PATCH'
+        ]);
+        $request = ServerRequestFactory::fromGlobals(['REQUEST_URI' => '/articles-patch'], null, ['_method' => 'PATCH']);
```

Comment:
Updated, I also changed `RoutingMiddleware` to update the request object using `withMethod` is that alright?